indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/symmetric/SymmetricKey.kt

```diff
@@ -0,0 +1,211 @@
+                algorithm, secretKey
+            ), SymmetricKey.RequiringNonce<AuthCapability.Authenticated.Integrated, KeyType.Integrated>
+
+            class WithoutNonce(
+                algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Authenticated.Integrated, NonceTrait.Without, KeyType.Integrated>,
+                secretKey: ByteArray,
+                override val additionalProperties: MutableMap<String, String> = mutableMapOf<String, String>()
+            ) : Authenticating<NonceTrait.Without>(
+                algorithm, secretKey
+            ), SymmetricKey.WithoutNonce<AuthCapability.Authenticated.Integrated, KeyType.Integrated>
+        }
+
+        sealed class NonAuthenticating<I : NonceTrait>(
+            algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Unauthenticated, I, KeyType.Integrated>,
+            secretKey: ByteArray
+        ) : Integrated<AuthCapability.Unauthenticated, I>(algorithm, secretKey), SymmetricKey.NonAuthenticating<I> {
+            class RequiringNonce(
+                algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Unauthenticated, NonceTrait.Required, KeyType.Integrated>,
+                secretKey: ByteArray,
+                override val additionalProperties: MutableMap<String, String> = mutableMapOf<String, String>()
+            ) : NonAuthenticating<NonceTrait.Required>(algorithm, secretKey)
+
+            class WithoutNonce(
+                algorithm: SymmetricEncryptionAlgorithm<AuthCapability.Unauthenticated, NonceTrait.Without, KeyType.Integrated>,
+                secretKey: ByteArray,
+                override val additionalProperties: MutableMap<String, String> = mutableMapOf<String, String>()
+            ) : NonAuthenticating<NonceTrait.Without>(algorithm, secretKey)
+        }
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other !is Integrated<*, *>) return false
+
+            if (algorithm != other.algorithm) return false
+            if (!secretKey.contentEquals(other.secretKey)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = algorithm.hashCode()
+            result = 31 * result + secretKey.contentHashCode()
+            return result
+        }
+    }
+
+
+    /**
+     * Encryption key with dedicated MAC key. Used for non-authenticated ciphers that use an external MAC function to
+     * bolt on AEAD capabilities, such as [SymmetricEncryptionAlgorithm.AES.GCM]
+     */
+    sealed class WithDedicatedMac<I : NonceTrait>
+    @HazardousMaterials("Does not check whether key size matched algorithm! Useful for testing, but not production!")
+    /**
+     * Do not invoke directly! use Supreme's `SymmetricEncryptionAlgorithm.randomKey()` and `SymmetricEncryptionAlgorithm.encryptionKeyFrom(bytes)`
+     * This constructor does not check for matching key sizes to allow for testing error cases!
```

Comment:
same as above; a public constructor is the natural way to construct a class. kdoc is not the appropriate way to discourage someone from using this natural way.

if you don't want it to be used, don't make it public.

Reply:
and gone. this was a noversight. at one point i had it in the tests of another module